less/data_selection/get_validation_dataset.py

```diff
@@ -296,6 +306,54 @@ def format_example(df, idx, include_answer=True):
     return dataset
 
 
+def get_kstack_dataset(data_dir: str,
+                       tokenizer: PreTrainedTokenizerBase,
+                       max_length: int,
+                       use_chat_format=False,
+                       chat_format="tulu",
+                       **kwargs):
+
+
+    data = load_dataset("json",
+                        data_files=[f"{data_dir}/eval/kstack/validation.jsonl"]
+                    )["train"]
+
+    tokenized_data = data.map(lambda example: tokenize_dataset_with_content(example, tokenizer, max_length), batched=False, num_proc=6, remove_columns=data.column_names)
+    return tokenized_data
+
+
+def get_kstack_clean_dataset(data_dir: str,
+                             tokenizer: PreTrainedTokenizerBase,
+                             max_length: int,
+                            use_chat_format=False,
+                            chat_format="tulu",
+                            **kwargs):
+    data = load_dataset("JetBrains/KStack-clean")["train"]
+    tokenized_data = data.map(lambda example: tokenize_dataset_with_content(example, tokenizer, max_length), batched=False, num_proc=6, remove_columns=data.column_names)
+    return tokenized_data
+
+
+def get_golden_repos(data_dir: str,
+                     tokenizer: PreTrainedTokenizerBase,
+                     max_length: int,
+                    use_chat_format=False,
+                    chat_format="tulu",
+                    **kwargs):
+    data = load_from_disk(f"{data_dir}/eval/golden_repos")
+    tokenized_data = data.map(lambda example: tokenize_dataset_with_content(example, tokenizer, max_length), batched=False, num_proc=6, remove_columns=data.column_names)
+    return tokenized_data
+
+def get_lca_no_context(data_dir: str,
+                       tokenizer: PreTrainedTokenizerBase,
+                       max_length: int,
+                      use_chat_format=False,
+                      chat_format="tulu",
+                      **kwargs):
+      data = load_from_disk(f"{data_dir}/eval/lca_no_context")
+      tokenized_data = data.map(lambda example: tokenize_dataset_with_content(example, tokenizer, max_length), batched=False, num_proc=6, remove_columns=data.column_names)
+      return tokenized_data
+
+
```

Comment:
Maybe we can unify this to one or two functions which either load dataset from disk or from HF (and we'll probably have to add s3 later), putting the full path in 'data_dir'?

Reply:
Implemented it, tbh it didn't reduce the length of this code block significantly, because we have 4 datasets and 3 load methods (local HF dataset, remote HF dataset, local json file). But might be worth it if we add more options 

JFIY I also edited `matching.py` to get the number of tasks automatically instead of specifying them manually to also try to reduce the amount of editing you need to apply when adding a custom dataset